enzy_htp/preparation/reactive_docking.py

```diff
@@ -0,0 +1,1285 @@
+"""Driver for the reactive docking functionality available in enzy_htp. The only function that that should be 
+called is dock_reactants(). All others are implementation functions that SHOULD NOT be used. 
+
+Author: Chris Jurich <chris.jurich@vanderbilt.edu>
+Date: 2023-07-28
+"""
+import os
+from pathlib import Path
+from collections import defaultdict
+from typing import List, Tuple, Set
+from copy import deepcopy
+
+import numpy as np
+import pandas as pd
+
+from enzy_htp import interface, config, _LOGGER, binding_energy
+from enzy_htp._interface.rosetta_interface import RosettaCst
+
+import enzy_htp.chemical as chem
+from enzy_htp import mutation as mm
+from enzy_htp.structure import PDBParser, Structure
+from enzy_htp.core import file_system as fs
+
+#TODO(CJ): I should go through and use existing values if cache mode is selected
+#TODO(CJ): need to be able to specify the size of the grid
+
+
+def dock_reactants(structure: Structure,
+                   reactants: List[str],
+                   constraints: List[RosettaCst] = None,
+                   mutations: List[List[mm.Mutation]] = None,
+                   work_dir: str = None,
+                   save_work_dir: bool = True,
+                   reactant_conformers: List[str] = None,
+                   rng_seed: int = 1996,
+                   n_struct: int = 1000,
+                   cst_cutoff: int = 10,
+                   clash_distance: float = 2.0,
+                   clash_cutoff: int = 1,
+                   binding_cutoff: int = 20,
+                   cluster_distance: float = 2.5,
+                   use_cache: bool = True,
+                   parent: bool = False,
+                   pdb_code: None = str) -> List[str]:
```

Comment:
It will be great if the output can be a Structure() or list of them

Reply:
I agree. For now I was just doing everything in files but I will change the output format eventually.